### ISSUE TITLE
zhuying - Identical withdraw request Id may prevent accounts from deleveraging

### ISSUE THREAD

**Issue by sherlock-admin3** [sherlock-audit/2024-06-leveraged-vaults-judging#25]
zhuying

Medium

# Identical withdraw request Id may prevent accounts from deleveraging

## Summary
Withdraw request Id may be identical and accounts with same withdraw request Id can't deleverage.
## Vulnerability Detail
```solidity
            SplitWithdrawRequest memory s = VaultStorage.getSplitWithdrawRequest()[w.requestId];
            // Safety check to ensure that the split withdraw request is not active, split withdraw
            // requests are never deleted. This presumes that all withdraw request ids are unique.
            require(s.finalized == false && s.totalVaultShares == 0);
```
Code comments show "This presumes that all withdraw request ids are unique".  However there is a chance that withdraw request Id could be the same. Let me explain:
For PendlePTKelpVault and PendlePTStakedUSDEVault, the withdraw request Id is uint256 of address of `CooldownHolder`:
```solidity
        return uint256(uint160(address(holder)));
```
For PendlePTEtherFiVault, the withdraw request Id is the Id of [Withdraw Request NFT](https://etherscan.io/token/0x7d5706f6ef3f89b3951e23e557cdfbc3239d4e2c):
```solidity
        return LiquidityPool.requestWithdraw(address(this), eETHReceived);
```
This dune query(https://dune.com/queries/3874537) shows the range of Withdraw Request NFT Ids is approximately from 500+ to 20,000+ and the Ids are continuously incremented.
So there is a possibility that the uint256 of address of `CooldownHolder` could be identical to Withdraw Request NFT Id.
## Impact
The impact is:
1. Account A request withdraw and get a requestId.
2. Account A deleverage his account. Split withdraw requests are never deleted. So `totalVaultShares` != 0.
```solidity
         VaultStorage.getSplitWithdrawRequest()[w.requestId].totalVaultShares = w.vaultShares;
```
4. Account B request withdraw and get a requestId (same as Account A).
5. Account B can't deleverage his account (require `totalVaultShares` == 0).
```solidity
            SplitWithdrawRequest memory s = VaultStorage.getSplitWithdrawRequest()[w.requestId];
            require(s.finalized == false && s.totalVaultShares == 0);
```
In conclusion accounts with same withdraw request Id can't deleverage.
## Code Snippet
https://github.com/sherlock-audit/2024-06-leveraged-vaults/blob/main/leveraged-vaults-private/contracts/vaults/common/WithdrawRequestBase.sol#L211-L214
https://github.com/sherlock-audit/2024-06-leveraged-vaults/blob/main/leveraged-vaults-private/contracts/vaults/staking/protocols/Ethena.sol#L103-L112
https://github.com/sherlock-audit/2024-06-leveraged-vaults/blob/main/leveraged-vaults-private/contracts/vaults/staking/protocols/EtherFi.sol#L24-L28
https://github.com/sherlock-audit/2024-06-leveraged-vaults/blob/main/leveraged-vaults-private/contracts/vaults/staking/protocols/Ethena.sol#L103-L112
## Tool used
manual
## Recommendation
No good idea.
